R/fread.R

```diff
@@ -267,8 +267,14 @@ yaml=FALSE, autostart=NA, tmpdir=tempdir())
     if (is.integer(skip)) skip = skip + n_read
   }
   warnings2errors = getOption("warn") >= 2
+  stopifnot(identical(tz,"UTC") || identical(tz,""))
+  if (tz=="") {
+    tt = Sys.getenv("TZ", unset=NA_character_)
```

Comment:
From `?Sys.timezone`

> As from R 3.5.0, when a time zone location is first found in a session, its value is cached in object .sys.timezone in the base environment.

Should we check this as well? [1] check `.sys.timezone` [2] check `Sys.getenv('TZ')`.

Not sure it's worth getting that far into the weeds.

Reply:
All I've ever seen in use, especially production use, is the TZ environment variable. I think that's universal and standard across languages and OS, afaik. I don't even know what a "time zone location" is, but it sounds like something you'd need if we need to distinguish different local time zones, which we don't need. All we need is UTC or not-UTC. But anyway, we don't want the cached value at the start of the R session; if a user enters `Sys.setenv(TZ="UTC")`, as we now do in tests, then we want to recognize that latest value.

Reply:
And it comes back to the strangeness that if I do `Sys.setenv(TZ="UTC")`, `Sys.timezone()` isn't affected.  How then, do I change the value of `Sys.timezone()` if it isn't via the standard `TZ` envornment variable?!

Reply:
ok makes sense to me

Reply:
maybe related to the caching thing within session